base/loggers/OwnSplitChannel.cpp

```diff
@@ -92,9 +92,13 @@ void OwnSplitChannel::logSplit(const Poco::Message & msg)
 
         elem.source_line = msg.getSourceLine();
 
-        std::lock_guard<std::mutex> lock(text_log_mutex);
+        std::unique_lock<std::mutex> lock(text_log_mutex);
         if (auto log = text_log.lock())
-            log->add(elem);
+        {
+            lock.unlock();
+            log->add(std::move(elem));
+        }
+
```

Comment:
I can already imagine how I fail to notice the `lock.unlock()` inside the `if`, add some code after `if` and believe that it's under the lock. Let's restructure:
```
shared_ptr<TextLog> text_log_locked;
{
    std::lock_guard<std::mutex> lock(text_log_mutex);
    text_log_locked = text_log.lock();
}
if (text_log_locked)
{
   text_log_locked->add(elem);
}
```